crates/primitives/src/l1.rs

```diff
@@ -983,48 +983,30 @@ impl<'a> Arbitrary<'a> for TaprootSpendPath {
     }
 }
 
-/// [Borsh](borsh)-friendly Bitcoin [`Transaction`].
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct BitcoinTx(Transaction);
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+pub struct RawBitcoinTx(Vec<u8>);
 
-impl From<Transaction> for BitcoinTx {
+impl From<Transaction> for RawBitcoinTx {
     fn from(value: Transaction) -> Self {
-        Self(value)
-    }
-}
-
-impl From<BitcoinTx> for Transaction {
-    fn from(value: BitcoinTx) -> Self {
-        value.0
+        Self(serialize(&value))
     }
 }
 
-/// Implement BorshSerialize using Bitcoin consensus serialization.
-impl BorshSerialize for BitcoinTx {
-    fn serialize<W: Write>(&self, writer: &mut W) -> std::io::Result<()> {
-        // Use bitcoin's consensus serialization
-        let tx_bytes = serialize(&self.0);
-        BorshSerialize::serialize(&(tx_bytes.len() as u32), writer)?;
-        writer.write_all(&tx_bytes)
+impl TryFrom<RawBitcoinTx> for Transaction {
+    type Error = bitcoin::consensus::encode::Error;
+    fn try_from(value: RawBitcoinTx) -> Result<Self, Self::Error> {
+        deserialize(&value.0)
     }
 }
 
-/// Implement BorshDeserialize using Bitcoin consensus deserialization.
-impl BorshDeserialize for BitcoinTx {
-    fn deserialize_reader<R: Read>(reader: &mut R) -> std::io::Result<Self> {
-        // First, read a Vec<u8> using Borsh (this picks up the length)
-        let tx_len = u32::deserialize_reader(reader)? as usize;
-        let mut tx_bytes = vec![0u8; tx_len];
-        reader.read_exact(&mut tx_bytes)?;
-
-        // Now parse those bytes with bitcoin consensus
-        let tx = deserialize(&tx_bytes).map_err(|err| Error::new(ErrorKind::InvalidData, err))?;
-
-        Ok(BitcoinTx(tx))
+impl TryFrom<&RawBitcoinTx> for Transaction {
+    type Error = bitcoin::consensus::encode::Error;
```

Comment:
import this above?

```rs
use bitcoin::consensus
```
then

```suggestion
    type Error = consensus::encode::Error;
```